Classes/ios/Scanners/MTBBarcodeScanner.m

```diff
@@ -716,17 +722,18 @@ - (BOOL)isCapturingStillImage {
 
 #pragma mark - Setters
 
-- (void)setCamera:(MTBCamera)camera {
-    [self setCamera:camera error:nil];
-}
+- (BOOL)setCamera:(MTBCamera)camera error:(NSError **)error {
+    AVCaptureDeviceInput *input;
 
-- (void)setCamera:(MTBCamera)camera error:(NSError **)error {
     if (self.isScanning && camera != _camera) {
         AVCaptureDevice *captureDevice = [self newCaptureDeviceWithCamera:camera];
-        AVCaptureDeviceInput *input = [self deviceInputForCaptureDevice:captureDevice error:error];
+        input = [AVCaptureDeviceInput deviceInputWithDevice:captureDevice error:error];
         [self setDeviceInput:input session:self.session];
     }
+
     _camera = camera;
```

Comment:
We should probably only set the camera ivar in case changing the input device was successful, so the property reflects the correct state.